src/main/java/uk/ac/ebi/ena/webin/cli/upload/FtpService.java

```diff
@@ -127,134 +118,57 @@ protected void finalize() {
-                    // continues to return true. So need to check the status here before attempting to connect again.
-                    connect();
-                }
+                ftpClient.connect(SERVER, FTP_PORT);
+                ftpClient.enterLocalPassiveMode();
 
                 if (!ftpClient.login(username, password)) {
                     logErrorFtpReply();
-                    throw new FTPAuthenticationFailureException();
-                }
-
-                return null;
-            }, context -> log.warn("Retrying FTP server login."), IOException.class, FTPAuthenticationFailureException.class);
-        } catch (FTPAuthenticationFailureException e) {
-            throw WebinCliException.userError(WebinCliMessage.SERVICE_AUTHENTICATION_ERROR.format("FTP"));
-        } catch (Exception e) {
-            throw WebinCliException.systemError(e, WebinCliMessage.FTP_SERVER_ERROR.text());
-        }
-    }
-
-    private void setFileTypeAsBinary() {
-        try {
-            RetryUtils.executeWithRetry((RetryCallback<Void, Exception>) context -> {
-                if (context.getLastThrowable() != null && context.getLastThrowable() instanceof IOException) {
-                    connect();
-                    login();
+                    throw WebinCliException.userError(WebinCliMessage.SERVICE_AUTHENTICATION_ERROR.format("FTP"));
                 }
 
                 if( !ftpClient.setFileType( FTP.BINARY_FILE_TYPE ) ) {
                     logErrorFtpReply();
                     throw WebinCliException.systemError(WebinCliMessage.FTP_SERVER_ERROR.text());
                 }
 
-                return null;
-            }, context -> log.warn("Retrying setting file type on FTP server."), IOException.class);
-        } catch (WebinCliException e) {
-            throw e;
-        } catch (Exception e) {
-            throw WebinCliException.systemError(e, WebinCliMessage.FTP_SERVER_ERROR.text());
-        }
-    }
-
-    private void changeToLastWorkingDirectory() {
-        if (lastWorkingDirectory == null) {
-            return;
-        }
-
-        try {
-            RetryUtils.executeWithRetry((RetryCallback<Void, Exception>) context -> {
-                if (context.getLastThrowable() != null && context.getLastThrowable() instanceof IOException) {
-                    connect();
-                    login();
-                    setFileTypeAsBinary();
-                }
-
-                if( !ftpClient.changeWorkingDirectory(lastWorkingDirectory)) {
-                    logErrorFtpReply();
-                    throw WebinCliException.systemError(WebinCliMessage.FTP_CHANGE_DIR_ERROR.format(lastWorkingDirectory));
+                // Upon reconnect change FTP server's working back to what it was before.
+                if (ftpServerWorkingDir != null) {
+                    if( !ftpClient.changeWorkingDirectory(ftpServerWorkingDir.toString().replaceAll("\\\\+", "/"))) {
```

Comment:
Might be nice to have a file path normalization method (toLinuxPath) that coverts Windows paths to Linux.